### ISSUE TITLE
"ARIA toggle fields have an accessible name" error for DropDownList with autoBind false

### ISSUE THREAD

**Issue by Dimitar-Goshev** [telerik/kendo-ui-core#6462]
### Bug report
"Ensures every ARIA toggle field has an accessible name" error for DropDownList with autoBind false.
In this scenario the DropDownList has a selected option element with no text, causing the above accessibility error thrown by Axe.

### Reproduction of the problem
1. Run the following [full screen Dojo](https://runner.telerik.io/fullscreen/@Dimitar-Goshev/eVePUPaT) .
2. Open Chrome dev tools and run Axe dev tools scan
3. 

### Current behavior
"Ensures every ARIA toggle field has an accessible name" error for span without text or area-label:
`<span id="eb087130-28bb-486c-bd00-30aa249fabbd" unselectable="on" role="option" aria-selected="true" class="k-input"></span>`

### Expected/desired behavior
No accessibility error.

### Environment
* **Kendo UI version:** 2021.2.616
* **Browser:** all

**Comment by Dimitar-Goshev** [comment]
This is a false-positive issue:
[Dojo examples](https://dojo.telerik.com/@Dimitar-Goshev/AditIyib), analyzed with Chrome axe DevTools 4.2.3 on the Dojo's [full screen link](https://runner.telerik.io/fullscreen/@Dimitar-Goshev/AditIyib).

Case 2 - "ARIA toggle fields have an accessible name" is thrown when the DropDownList initialization input element is wrapped inside the label element. This error is reported by axe as an issue which "Needs Review". The error is reported for a span element with role="option". The "[more info](https://dequeuniversity.com/rules/axe/4.2/aria-toggle-field-name?application=AxeChrome)" link in axe about the error explains that this rule is relevant for the following semantic roles:

-     checkbox
-     menu
-     menuitemcheckbox
-     menuitemradio
-     radio
-     radiogroup
-     switch

Since our span element does not have any of the above roles (it has role="option"), then why is this error flagged. This is because the [role option](https://www.w3.org/TR/wai-aria-1.2/#option) has a supported state aria-checked by specification. The [area-checked](https://www.w3.org/TR/wai-aria-1.2/#aria-checked) state is evaluated by axe as a possible toggle field, as it is used in also in _checkbox, menuitemcheckbox, radio and switch_ - all listed in the toggle list above. But our span element with role="option" is **not using aria-checked**, it is using **aria-selected only**. So we consider this as a **false-positive** issue. Still, for DropDownList we do not recommend the setup where the initialization element is wrapped inside a label element.

Case 3 - Label for setup and autoBind: true by default - this setup reports no error and we would recommend this setup:

<label for="products2">3. Label for - Recommended for DropDownList</label>
<input id="products2" style="width: 100%;" />

Case 4 - Label for setup and **autoBind: false** - this setup throws "ARIA toggle fields have an accessible name" issue. Having in mind the explanation for false-positive issue for the toggle fields report form Case 2 above, we consider this error also as false-positive report. I tested the test page with Firefox and **NVDA screen reader and the respective label for this DropDownList is read when focusing the component**. This is also the case for cases 2 and 3.
